patroni/__main__.py

```diff
@@ -164,10 +165,23 @@ def patroni_main(configfile: str) -> None:
 
 def process_arguments() -> Namespace:
     parser = get_base_arg_parser()
-    parser.add_argument('--validate-config', action='store_true', help='Run config validator and exit')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--validate-config', action='store_true', help='Run config validator and exit')
+    group.add_argument('--generate-sample-config', action='store_true',
+                       help='Generate a sample Patroni yaml configuration file')
+    group.add_argument('--generate-config', action='store_true',
+                       help='Generate a Patroni yaml configuration file for a running instance')
+    parser.add_argument('--dsn', help='Optional DSN string of the instance to be used as a source \
+                                    for config generation. Superuser connection is required.')
```

Comment:
I feel like we should start creating some sub-commands in this CLI instead of creating several different flags. IDK, some random idea:

In the main `patroni` command we could have a `-c/--config-file` flag that would apply to all sub-commands. Then we could have some sub-commands like:

* `patroni run` to run the agent. Requires `-c` to be set.
* `patroni config`: to group config related commands:
    * `patroni config validate`: to validate a configuration file. Requires `-c` to be set.
    * `patroni config generate`: to generate a configuration file. If `-c` is set, write to that file, otherwise write to stdout. It could have a couple flags:
        * `--sample`: if the user wants to create a sample file. If not given would require connecting to a running instance.
        * `--dsn`: optional DSN to connect to the running instance. Mutually exclusive with `--sample`.

What do you think?

Reply:
even if we find a way to implement it backward-compatible, let's not do it in this PR:)

Reply:
These were just some side thoughts because I felt we are starting to pile up commands through flags, and personally I think we should adjust things sooner than later, otherwise the impact might be higher later 😆 

There is the counter part of existing deployments, yes. Maybe increasing a more significant digit in the version when releasing could pull some attention that the upgrade may break things. I mean, that's what in see in Postgres. We have the minor releases that in theory should not break things (although I already saw a minor breaking the application once 😆 ), and the major releases that have more structural changes that introduce new things and breaking changes.

I guess this goes more or less in the same direction of some thoughts that came up in Essen during the meeting.